components/script/dom/htmldialogelement.rs

```diff
@@ -70,6 +70,31 @@ impl HTMLDialogElementMethods for HTMLDialogElement {
         *self.return_value.borrow_mut() = return_value;
     }
 
+    // https://html.spec.whatwg.org/multipage/#dom-dialog-show
+    fn Show(&self) {
+        let element = self.upcast::<Element>();
+
+        // Step 1 TODO: Check is modal flag is false
+        if element.has_attribute(&local_name!("open")) {
```

Comment:
This, without the other check, seems like Step 2 to me, which could already be implemented instead.

Reply:
It's not, (admittedly the spec isn't super clear here) step 2 throws an exception if you try and call .show() on an already open modal dialog. Step 1 is an early return for calling .show() on an already open non-modal dialog.